### PR TITLE
slxos plugin implementation

### PR DESCRIPTION
**Pull request by fpessoanunes**
Extreme SLXOS plugin Implementation

### CODE REVIEW

File: src/unicon/plugins/slxos/statemachine.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        config = State('config', patterns.config_prompt)
+
+        ##########################################################
+        # Path Definition
+        ##########################################################
+        enable_to_config = Path(enable, config, 'configure', None)
+        config_to_enable = Path(config, enable, 'exit', None)
+
+        # Add State and Path to State Machine
+        self.add_state(enable)
```
Comment (tahigash) [inline src/unicon/plugins/slxos/statemachine.py:43]: no non-enable mode?

File: src/unicon/plugins/slxos/settings.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    def __init__(self):
+        super().__init__()
+        self.HA_INIT_EXEC_COMMANDS = [
+            'terminal length 0',
+            'terminal timeout 0',
+            'no terminal monitor',
+            'show version'
+        ]
+        self.CONNECTION_TIMEOUT = 60*5
```
Comment (tahigash) [inline src/unicon/plugins/slxos/settings.py:26]: I thought 5 mins by default is a bit long. is it appropriate for this nos?

File: src/unicon/plugins/slxos/statemachine.py (hunk 1 above)
Comment (fpessoanunes) [reply src/unicon/plugins/slxos/statemachine.py:43]: there is no non-enable mode in this nos

File: src/unicon/plugins/slxos/settings.py (hunk 2 above)
Comment (fpessoanunes) [reply src/unicon/plugins/slxos/settings.py:26]: ok, I believe we can leave the default inherited from GenericSettings

File: src/unicon/plugins/slxos/statemachine.py (hunk 1 above)
Comment (tahigash) [reply src/unicon/plugins/slxos/statemachine.py:43]: thanks for the update!

File: src/unicon/plugins/slxos/settings.py (hunk 2 above)
Comment (tahigash) [reply src/unicon/plugins/slxos/settings.py:26]: ok. default is 60 secs. so if you agree, can you remove this line here?

**Review by tahigash — APPROVED** [review]
approved with 1 comment.

**Review by LukasMcClelland — COMMENTED** [review]
Thank you for the excellent contribution! Left a few comments

File: src/unicon/plugins/slxos/settings.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    def __init__(self):
+        super().__init__()
+        self.HA_INIT_EXEC_COMMANDS = [
+            'terminal length 0',
+            'terminal timeout 0',
+            'no terminal monitor',
+            'show version'
+        ]
+        self.HA_INIT_CONFIG_COMMANDS = []
```
[the remark this answers is not in the document]
Comment (fpessoanunes) [reply src/unicon/plugins/slxos/settings.py:26]: None of these default commands apply to slxos. No config commands.

File: src/unicon/plugins/tests/test_plugin_slxos.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                        init_exec_commands=[],
+                        init_config_commands=[]
+                        )
+        c.connect()
+        cmd = 'configure'
+        ret = c.execute(cmd).replace('\r', '')
+        self.assertIn("{hostname}(config)#".format(hostname=hostname), c.spawn.match.match_output)
+        cmd = 'end'
+        ret = c.execute(cmd).replace('\r', '')
+        self.assertIn("{hostname}#".format(hostname=hostname), c.spawn.match.match_output)
```
Comment (d-the-iii) [inline src/unicon/plugins/tests/test_plugin_slxos.py:71]: What is the purpose of ret in this test case? In the previous one it was used in the assert, but not here.

**Review by d-the-iii — COMMENTED** [review]
LGTM, just one question is all.

File: src/unicon/plugins/slxos/service_implementation.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class Copy(Copy):
+
+    def __init__(self, connection, context, **kwargs):
+        super().__init__(connection, context, **kwargs)
+        self.start_state = 'enable'
+        self.end_state = 'enable'
+        self.dialog += Dialog([slxos_statements.save_confirm])
+        self.error_pattern = [r'Error while parsing source URL\. Please refer the usage and enter the complete command\.']
```
[the remark this answers is not in the document]
Comment (fpessoanunes) [reply src/unicon/plugins/slxos/service_implementation.py]: It was not a typo, however I noticed that error_pattern from Generic already covers the slxos. Removed it.

File: src/unicon/plugins/tests/test_plugin_slxos.py (hunk 4 above)
Comment (fpessoanunes) [reply src/unicon/plugins/tests/test_plugin_slxos.py]: no purpose, removed it.

**Review by d-the-iii — APPROVED** [review]

**Review by LukasMcClelland — APPROVED** [review]
Looks good. Thank you for addressing comments!

**Pull request merged**